src/helper.ts

```diff
@@ -383,6 +386,110 @@ export function updateLiquidatedPosition(
+  const txLogs = receipt ? receipt.logs : [];
+
+  const id = txHash + "-" + otoken;
+
+  const optionsBoughtAction = new OptionsBoughtAction(id);
+
+  optionsBoughtAction.otoken = otoken;
+  optionsBoughtAction.buyer = buyer;
+  optionsBoughtAction.amount = amount;
+  optionsBoughtAction.premium = premium;
+  optionsBoughtAction.fee = fee;
+  optionsBoughtAction.timestamp = timestamp;
+  optionsBoughtAction.transactionHash = txHash;
+
+  optionsBoughtAction.save();
+
+  const total = premium.plus(fee);
+
+  // add fees to stats
+  updateStats(amount, fee, true);
+
+  for (let i = 0; i < txLogs.length; ++i) {
+    // if event is to Controller, avoid reading all events
+    if (txLogs[i].address.toHexString() == CONTROLLER) {
+      // if topic is ShortOtokenBurned and account owner is tx sender (trader)
+      if (
+        txLogs[i].topics[0].toHexString() == SHORT_OTOKEN_BURNED &&
+        txLogs[i].topics[2].toHexString().slice(26) ==
+          from.toHexString().slice(2)
+      ) {
+        // if topic is shortOTokenBurned and account owner is tx sender (trader)
+        updateOptionShortPosition(true, buyer, otoken, amount, id, total);
+        return;
+      }
+    }
+  }
+  updateOptionLongPosition(true, buyer, otoken, amount, id, total);
+}
+
+export function addOptionsSoldAction(
+  event: OptionsSold,
+): void {
+
+  const txHash = event.transaction.hash.toHex();
+
+  const receipt = event.receipt;
+  const txLogs = receipt ? receipt.logs : [];
+
+  const seller = event.params.seller;
+  const otoken = event.params.series.toHex();
+  const amount = event.params.optionAmount;
+
+  const id = txHash + "-" + otoken;
+
+  const optionsSoldAction = new OptionsSoldAction(id);
+
+  optionsSoldAction.otoken = otoken;
+  optionsSoldAction.seller = seller;
+  optionsSoldAction.amount = amount;
+  optionsSoldAction.premium = event.params.premium;
+  optionsSoldAction.fee = event.params.fee;
+  optionsSoldAction.timestamp = event.block.timestamp;
+  optionsSoldAction.transactionHash = event.transaction.hash.toHex();
+
+  optionsSoldAction.save();
+
+  const total = event.params.premium.minus(event.params.fee);
+
+  // add fees to stats
+  updateStats(event.params.optionAmount, event.params.fee, false);
+
+  for (let i = 0; i < txLogs.length; ++i) {
+    // if event is to Controller, avoid reading all events
+    if (txLogs[i].address.toHexString() == CONTROLLER) {
+      // if topic is ShortOtokenMinted and account owner is tx sender (trader)
+      if (
+        txLogs[i].topics[0].toHexString() == SHORT_OTOKEN_MINTED &&
+        txLogs[i].topics[2].toHexString().slice(26) ==
+          event.transaction.from.toHexString().slice(2)
+      ) {
+        // if topic is shortOtokenMinted and account owner is tx sender (trader)
+        updateOptionShortPosition(false, seller, otoken, amount, id, total);
+        return;
+      }
+    }
+  }
+  updateOptionLongPosition(false, seller, otoken, amount, id, total);
+  
+}
+
```

Comment:
This is a good refactor for readability, but it'd be better to merge these into one function. The `updateOptionLongPosition` and `updateOptionShortPosition` functions are being exported but never consumed anywhere else so you should just be able to move their logic to the bottom of these functions.